ogr/parsing.py

```diff
@@ -71,6 +71,98 @@ def __str__(self) -> str:
         repo_url_str += ")"
         return repo_url_str
 
+    def __repr__(self) -> str:
+        return str(self)
+
+    @staticmethod
+    def _prepare_url(potential_url: str) -> Optional[ParseResult]:
+        # Remove trailing '/' from 1-7 to ensure parsing works as expected.
```

Comment:
I know these comments were copied over from the old implementation, but I fail to understand them. What are the numbers about?

Reply:
I think those numbers mean the examples in the function docstring. (But I agree with the update.)